<launch>
	<arg name="sigverse_ros_bridge_port"        default="50001" />
	<arg name="ros_bridge_port"                 default="9090" />

	<node name="hsr_teleop_key" pkg="sigverse_hsr_teleop_key" type="hsr_teleop_key" output="screen" launch-prefix="gnome-terminal -t 'HSR teleop key' -- ">
		<!--
		<remap from="/hsrb/message/to_robot"                  to="/hsrb/message/to_robot" />
		<remap from="/hsrb/message/to_human"                  to="/hsrb/message/to_human" />
		<remap from="/hsrb/joint_states"                      to="/hsrb/joint_states" />
		<remap from="/hsrb/command_velocity"                  to="/hsrb/command_velocity" />
		<remap from="/hsrb/omni_base_controller/command"      to="/hsrb/omni_base_controller/command" />
		<remap from="/hsrb/arm_trajectory_controller/command" to="/hsrb/arm_trajectory_controller/command" />
		<remap from="/hsrb/gripper_controller/command"        to="/hsrb/gripper_controller/command" />
		<remap from="/hsrb/suction_control/goal"              to="/hsrb/suction_control/goal" />
		<remap from="/hsrb/suction_control/result"            to="/hsrb/suction_control/result" />
		-->
	</node>

	<group ns="sigverse_ros_bridge">
		<node name="sigverse_ros_bridge" pkg="sigverse_ros_bridge" type="sigverse_ros_bridge" args="$(arg sigverse_ros_bridge_port)"/>
	</group>
	
	<include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" > 
		<arg name="port" value="$(arg ros_bridge_port)"/>
	</include>
	
	<node name="suction_server" pkg="tmc_suction" type="suction_server">
		<remap from="/pressure_sensor_on" to="/hsrb/pressure_sensor_on" />
		<remap from="/suction_on"         to="/hsrb/suction_on" />
		<remap from="/suction_control"    to="/hsrb/suction_control" />
	</node>
</launch>

